tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py

```diff
@@ -57,11 +57,21 @@ def get_recording_id():
 def get_test_id():
     # type: () -> str
     # pytest sets the current running test in an environment variable
+    # the path to the test can depend on the environment, so we can't assume this is the path from the repo root
     setting_value = os.getenv("PYTEST_CURRENT_TEST")
-
     path_to_test = os.path.normpath(setting_value.split(" ")[0])
-    path_components = path_to_test.split(os.sep)
-
+    full_path_to_test = os.path.abspath(path_to_test)
+
+    # walk up to the repo root by looking for "sdk" directory or root of file system
+    path_components = []
+    head, tail = os.path.split(full_path_to_test)
+    while tail != "sdk" and tail != "":
```

Comment:
I don't _think_ you have any other stop cases here. Famous last words 👍

Reply:
Writing a `while` loop always terrifies me, but I'm also _pretty_ sure this should be good. Knocking furiously on wood